paddle/go/cclient/cclient.go

```diff
@@ -83,29 +84,46 @@ func cArrayToSlice(p unsafe.Pointer, len int) []byte {
 	return (*[1 << 30]byte)(p)[:len:len]
 }
 
+type selector bool
+
+func (s selector) Select() bool {
+	return bool(s)
```

Comment:
I thought this is clear :) Because the underlying type for `selector` is `bool`.
I don't fully understand how to do type switch here, could you paste the code here?